plugins/sto/sto.cpp

```diff
@@ -23,71 +28,56 @@ static void inline u8(std::wstring &dst, const std::string &src) {
 }
 
 static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, float *camera_pos, float *camera_front, float *camera_top, std::string &context, std::wstring &identity) {
-	char identblock[0x200];
-	char contextblock[0x80];
-	float posblock[64];
+	for (int i=0;i<3;i++)
+		avatar_pos[i] = avatar_front[i] = avatar_top[i] = camera_pos[i] = camera_front[i] = camera_top[i] = 0.0f;
+
+	// Boolean value to check if game addresses retrieval is successful
+	bool ok;
+	// Char values for extra features
+	char version[17], identity_block[0x200], context_block[0x80];
+	// State
+	unsigned char state;
+
+	// Peekproc and assign game addresses to our containers, so we can retrieve positional data
+	ok = peekProc(pModule + state_offset, &state, 1)							&& // Magical state value: 1 when spawned.
```

Comment:
In the original version, this was a DWORD (i.e., uint32_t). I'd include <stdint.h> and use uint32_t instead.